genai/example_test.go

```diff
@@ -361,6 +361,25 @@ func ExampleTool() {
 	printResponse(res)
 }
 
+func ExampleNewCallableFunctionDeclaration() {
+	// Define a function to use as a tool.
+	weatherToday := func(city string) string {
+		return "comfortable, if you have the right clothing"
+	}
+	// You can construct the Schema for this function by hand, or
+	// let Go reflection figure it out.
+	// This also makes the function automatically callable.
+	// Reflection can't see parameter names, so provide those too.
+	fd, err := genai.NewCallableFunctionDeclaration("CurrentWeather", "Get the current weather in a given location", weatherToday, "city")
+	if err != nil {
+		// Not every type can be used in a tool function.
+		panic(err)
+	}
+
+	// Use the FunctionDeclaration to populate Model.Tools.
```

Comment:
The example should probably be extended to show this?

Reply:
I started to do that. I can finish when I have an implementation.